### PR TITLE
Added the GH Issue number note to Writing Tests in Docs

### LINKED ISSUE #28703: Add Note About Referencing GH Issue Number to Test in Contrib Guide
We often ask people to add the GH Issue number as a comment to new tests, but that requirement isn't explicitly spelled out in the contributing guide. Might save some back and forth if that is added to the `Writing Tests` section:

https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#writing-tests

### PR DESCRIPTION
**Pull request by KiranAkadas**
- [x] closes #28703 
I've added the required note at the appropriate place in the documentation.

### CODE REVIEW

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -949,6 +949,11 @@ the expected correct result::
 
         assert_frame_equal(pivoted, expected)
 
+.. note::
+
+   Add the GitHub Issue Number as a Comment to the New Tests.
```
Comment (jreback) [inline doc/source/development/contributing.rst]: don't need to capitalizie new tests. Can you point to an example in the code, e.g.https://github.com/pandas-dev/pandas/blob/master/pandas/tests/series/test_apply.py#L166 (or you can just replicate this here maybe better)

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: Ok
I will change that capitalized word
and for the example - Shall i add the following line 
'For Example : # GH 20714 bug fixed in: GH 24275" 
or anything else?

Comment (jreback) [reply doc/source/development/contributing.rst]: yes that sounds good (make it clear its an example :->)

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: Or should i just replicate like
'For Example : def test_apply_categorical_with_nan_values(self, series):
                                 # GH 20714 bug fixed in: GH 24275
                                 s = pd.Series(series, dtype="category")
                                 result = s.apply(lambda x: x.split("-")[0])'

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: I'll be writing
For example - If you modify the test - "test_apply_categorical_with_nan_values" (GH 20714), please write-
def test_apply_categorical_with_nan_values(self, series):
# GH 20714 bug fixed in: GH 24275
s = pd.Series(series, dtype="category")

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: Is that fine? 
Should i add the entire function or a few lines is enough for example?

Comment (WillAyd) [reply doc/source/development/contributing.rst]: You can just add `GH XXXX` to the test that directly precedes this

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: Can you give me an example. I'm not getting it.
What exactly should i write as an example.
Sorry for pestering

Comment (WillAyd) [reply doc/source/development/contributing.rst]: See the test case in #28662 how it has # GH XXXX as a comment at the top of the test (you can find this in other PRs as well)

You can add that in the test case shown in this file

Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: I've added it in my latest commit.
Please let me know

File: doc/source/development/contributing.rst [hunk 2]
```diff
@@ -949,6 +949,15 @@ the expected correct result::
 
         assert_frame_equal(pivoted, expected)
 
+.. note::
+ 
+   Add the GitHub issue number as a comment to the new tests.
+   For Example - For a test solving issue number 20172, write  
+     | # GH 20172
```
Comment (WillAyd) [inline doc/source/development/contributing.rst]: Should move this to the function body. Check the test as an example in #28662 if needed

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by KiranAkadas** [comment]
Ok
I'm on it

File: doc/source/development/contributing.rst (hunk 1 above)
Comment (KiranAkadas) [reply doc/source/development/contributing.rst]: Please Check

**Comment by WillAyd** [comment]
Looks like this was superseded by #28907

**Pull request closed without merging**